cms/djangoapps/contentstore/views/course.py

```diff
@@ -1082,19 +1056,30 @@ def _modify_tabs_to_components(request, course_module):
             # user has failed to put iterable value into advanced component list.
             # return immediately and let validation handle.
             return
+        update_tab(course_tabs, tab_type, tab_enabled)
 
-        if tab_enabled:
-            # check passed, some of this component_types are present, adding tab
-            if _add_tab(request, tab_type, course_module):
-                # tab indeed was added, the change needs to propagate
-                tabs_changed = True
-        else:
-            # the tab should not be present (anymore)
-            if _remove_tab(request, tab_type, course_module):
-                # tab indeed was removed, the change needs to propagate
-                tabs_changed = True
+    # Additionally update any persistent tabs provided by course views
+    for tab_type in CourseTabManager.get_tab_types().values():
+        if getattr(tab_type, "is_persistent", False):
```

Comment:
The complication here is that the non-extension tab types are also returned, and so this code was trying to both detect the existence of the attribute and default it. Now that there is a base type I think it would be better to switch to using an ``issubtype`` test here instead (mirroring the previous test), and then calling the attribute normally as you suggest.